docs/Concise-Guide-for-Evaluating-Open-Source-Software.md

```diff
@@ -1,57 +1,117 @@
+| **Name Verification** | Check if a similar name is more popular - that could indicate a typosquatting attack. |   |
+| **Usage Metrics** | Assess if it has significant use. Software with many users or large users may be inappropriate for your use. However, widely-used software is more likely to offer useful information on how to use it securely, and more people will care about its security. |   |
+
+## Practical Testing
+
+| Rule | Description | Check |
+|------|-------------|:--------:|
+| **Behavior Testing** | Try adding the dependency as a test, preferably in an isolated environment. Does it exhibit malicious behaviour, e.g., does it attempt to exfiltrate sensitive data? |   |
+| **Dependency Impact** | Does it add unexpected or unnecessary indirect dependencies in production? For example, does it include production dependencies that are only required at development time or test time instead? If so, would their maintainers be willing to fix that? Every new dependency is a potential support problem or supply chain attack, so it's wise to eliminate unnecessary ones. |   |
+
+## Code Evaluation
+
+Even a brief review of software (by you, someone you hire, or someone else), along with recent changes to it, can give you some insight. Here are things to consider:
+
+| Rule | Description | Check |
+|------|-------------|:--------:|
+| **Code Completeness** | Evaluate whether there is evidence of insecure/incomplete software (e.g., many TODO statements). |   |
+| **Malicious Code Check** | Analyze if there is evidence that the software is malicious. Per [_Backstabber's Knife Collection_](https://arxiv.org/abs/2005.09535), check the installation scripts/routines for maliciousness, check for data exfiltration from **~/.ssh** and environment variables, and look for encoded/obfuscated values that are executed. Examine the most recent commits for suspicious code (an attacker may have added them recently). |   |
+| **Sandbox Testing** | Consider running the software in a sandbox to attempt to trigger and detect malicious code. |   |
+| **Security Implementations** | When reviewing its source code, is there evidence in the code that the developers were trying to develop secure software (such as rigorous input validation of untrusted input and the use of parameterized statements)? |   |
+| **Security Reviews** | See the [OpenSSF's list of security reviews](https://github.com/ossf/security-reviews/blob/main/Overview.md#readme). |   |
+| **Static Analysis** | What are the "top" problems reported by static analysis tools? |   |
+| **Test Validation** | Consider running all defined test cases to ensure the software passes them. |   |
+
+## Additional Resources
+
+- [The Tidelift guide to choosing packages well (February 2021)](https://tidelift.com/subscription/choosing-open-source-packages-well), Tidelift
+- [How to Evaluate Open Source Software / Free Software (OSS/FS) Programs](https://dwheeler.com/oss_fs_eval.html)
+
+
```

Comment:
```suggestion

```

You need to delete one of the lines to pass markdownlint.

Reply:
Should be fixed now

Reply:
Great! Which blank line did you delete :-) :-) ?